model/lifecycle.go

```diff
@@ -580,19 +596,15 @@ func CreateBuildFromVersion(args BuildCreateArgs) (string, error) {
 	// get a new build number for the build
 	buildNumber, err := db.GetNewBuildVariantBuildNumber(args.BuildName)
```

Comment:
I don't think we use the build number anywhere, but if the transaction fails we will have incremented the build number.

Reply:
Created https://jira.mongodb.org/browse/EVG-6785. If something were using this I think it would still be fine in that scenario